controllers/cluster_controller.go

```diff
@@ -164,6 +168,32 @@ func (r *ClusterReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cl
 	return res, kerrors.NewAggregate(errs)
 }
 
+func (r *ClusterReconciler) reconcileMetrics(_ context.Context, c *clusterv1.Cluster) {
+
+	if c.Status.ControlPlaneInitialized {
+		metrics.ClusterControlPlaneReady.WithLabelValues(c.Name, c.Namespace).Set(1)
+	} else {
+		metrics.ClusterControlPlaneReady.WithLabelValues(c.Name, c.Namespace).Set(0)
+	}
+	if c.Status.InfrastructureReady {
+		metrics.ClusterInfrastructureReady.WithLabelValues(c.Name, c.Namespace).Set(1)
+	} else {
+		metrics.ClusterInfrastructureReady.WithLabelValues(c.Name, c.Namespace).Set(0)
+	}
+	// TODO: [wfernandes] pass context here
+	_, err := secret.Get(r.Client, c, secret.Kubeconfig)
```

Comment:
I noticed that `secret.Get` is using `context.TODO` instead of being passed the `ctx` which in most cases is available. I thought that I could make a separate refactor PR for that.

Reply:
This is a good first issue for v1a3